### DISCUSSION TITLE
找不到windows kit

### DISCUSSION THREAD

**Discussion by congcong-hehe** [xmake-io/xmake#2568] [Q&A] [answered]
安装好xmake之后，创建了一个最简单的hello world c语言项目，使用xmake编译显示错误：
checking for platform ... windows
checking for architecture ... x64
checking for Microsoft Visual Studio (x64) version ... 2019
[ 25%]: ccache compiling.release src\main.c
error: main.c
src\main.c(1): fatal error C1083: 无法打开包括文件: “stdio.h”: No such file or directory

  > in src\main.c

推测原因是找不到windows kit，后面使用
xmake f --wdk="C:\Program Files (x86)\Windows Kits\10" -c后也没用。

使用xmake一段时间了，突然出现这样的问题，谢谢作者大大了

**Comment by waruqi** [comment]
xmake -vD 看下

**Reply by congcong-hehe** [reply]
PS D:\study_data\pro\mypro\test\xmake\hello> xmake -vD
checking for C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe ... ok
checking for flags (-Ox -fp:fast) ... ok
> cl.exe "-Ox" "-fp:fast"
checking for flags (-DNDEBUG) ... ok
> cl.exe "-DNDEBUG" "-nologo"
[ 25%]: ccache compiling.release src\main.c
"C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX64\\x64\\cl.exe" -c -nologo -Ox -fp:fast -DNDEBUG -Fobuild\.objs\hello\windows\x64\release\src\main.c.obj src\main.c
checking for flags (cl_sourceDependencies) ... ok
> cl.exe "/sourceDependencies" "C:\Users\11243\AppData\Local\Temp\.xmake\220715\_A73166C91C6743408BEE82051C822D50.json" "-nologo"
error: @programdir\modules\private\async\runjobs.lua:256: @programdir\modules\private\action\build\object.lua:85: @programdir\modules\core\tools\cl.lua:620: main.c
src\main.c(1): fatal error C1083: 无法打开包括文件: “stdio.h”: No such file or directory

stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:872]:
    [@programdir\modules\core\tools\cl.lua:620]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:127]: in function 'try'
    [@programdir\modules\core\tools\cl.lua:559]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:280]:
    [@programdir\modules\private\action\build\object.lua:85]: in function 'script'
    [@programdir\modules\private\action\build\object.lua:98]: in function 'build_object'
    [@programdir\modules\private\action\build\object.lua:123]: in function 'jobfunc'
    [@programdir\modules\private\async\runjobs.lua:232]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:280]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:121]: in function 'try'
    [@programdir\modules\private\async\runjobs.lua:218]: in function 'cotask'
    [@programdir\core\base\scheduler.lua:388]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:872: in function 'base/os.raiselevel'
        (...tail calls...)
        @programdir\modules\private\async\runjobs.lua:256: in field 'catch'
        @programdir\core\sandbox\modules\try.lua:127: in global 'try'
        @programdir\modules\private\async\runjobs.lua:218: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:388: in function <@programdir\core\base\scheduler.lua:385>

**Comment by waruqi** [comment]
```
xmake f -c -vD
xmake -rvD
```

再给我个完整 logs。。

**Reply by congcong-hehe** [reply]
PS D:\study_data\pro\mypro\test\xmake\hello> xmake f -c -vD
checking for platform ... windows
checking for architecture ... x64
checking for cl.exe ... C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe
checking for Microsoft Visual Studio (x64) version ... 2019
checking for dmd ... no
checkinfo: cannot runv(ldc2.exe --version), No such file or directory
checking for ldc2 ... no
checkinfo: cannot runv(gdc.exe --version), No such file or directory
checking for gdc ... no
checkinfo: cannot runv(zig.exe version), No such file or directory
checking for zig ... no
checkinfo: cannot runv(zig.exe version), No such file or directory
checking for zig ... no
checking for link.exe ... C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\link.exe
checking for the linker (ld) ... link.exe
configure
{
    vs = 2019
    clean = true
    mode = release
    buildir = build
    host = windows
    arch = x64
    kind = static
    ndk_stdcxx = true
    ccache = true
    plat = windows
}
PS D:\study_data\pro\mypro\test\xmake\hello> xmake -rvD
checking for cl.exe ... C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe
checking for the c compiler (cc) ... cl.exe
checking for C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64\cl.exe ... ok
checking for flags (-Ox -fp:fast) ... ok
> cl.exe "-Ox" "-fp:fast"
checking for flags (-DNDEBUG) ... ok
> cl.exe "-DNDEBUG" "-nologo"
[ 25%]: ccache compiling.release src\main.c
"C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Community\\VC\\Tools\\MSVC\\14.29.30133\\bin\\HostX64\\x64\\cl.exe" -c -nologo -Ox -fp:fast -DNDEBUG -Fobuild\.objs\hello\windows\x64\release\src\main.c.obj src\main.c
checking for flags (cl_sourceDependencies) ... ok
> cl.exe "/sourceDependencies" "C:\Users\11243\AppData\Local\Temp\.xmake\220715\_73ED93C25AD84C1088D53CE0683A9D60.json" "-nologo"
error: @programdir\modules\private\async\runjobs.lua:256: @programdir\modules\private\action\build\object.lua:85: @programdir\modules\core\tools\cl.lua:620: main.c
src\main.c(1): fatal error C1083: 无法打开包括文件: “stdio.h”: No such file or directory

stack traceback:
    [C]: in function 'error'
    [@programdir\core\base\os.lua:872]:
    [@programdir\modules\core\tools\cl.lua:620]: in function 'catch'
    [@programdir\core\sandbox\modules\try.lua:127]: in function 'try'
    [@programdir\modules\core\tools\cl.lua:559]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:280]:
    [@programdir\modules\private\action\build\object.lua:85]: in function 'script'
    [@programdir\modules\private\action\build\object.lua:98]: in function 'build_object'
    [@programdir\modules\private\action\build\object.lua:123]: in function 'jobfunc'
    [@programdir\modules\private\async\runjobs.lua:232]:
    [C]: in function 'xpcall'
    [@programdir\core\base\utils.lua:280]: in function 'trycall'
    [@programdir\core\sandbox\modules\try.lua:121]: in function 'try'
    [@programdir\modules\private\async\runjobs.lua:218]: in function 'cotask'
    [@programdir\core\base\scheduler.lua:388]:

stack traceback:
        [C]: in function 'error'
        @programdir\core\base\os.lua:872: in function 'base/os.raiselevel'
        (...tail calls...)
        @programdir\modules\private\async\runjobs.lua:256: in field 'catch'
        @programdir\core\sandbox\modules\try.lua:127: in global 'try'
        @programdir\modules\private\async\runjobs.lua:218: in upvalue 'cotask'
        @programdir\core\base\scheduler.lua:388: in function <@programdir\core\base\scheduler.lua:385>

**Comment by waruqi** [comment]
感觉你 vs 没装完整，用 vs 编译能过么。

执行下 `xmake l detect.sdks.find_vstudio` 贴下 logs

**Reply by congcong-hehe** [reply]
用VS是可以编译通过的

**Reply by congcong-hehe** [reply]
PS D:\study_data\pro\mypro\test\xmake\hello> xmake l detect.sdks.find_vstudio
{ 
  "2019" = { 
    vcvarsall = {
      x86 = {
        VSInstallDir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\",
        LIB = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\ATLMFC\lib\x86;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\lib\x86",
        VCToolsRedistDir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Redist\MSVC\14.29.30133\",
        INCLUDE = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\ATLMFC\include;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include",
        VS110COMNTOOLS = "C:\Program Files (x86)\Microsoft Visual Studio 11.0\Common7\Tools\",
        DevEnvdir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\",
        VisualStudioVersion = "16.0",
        ExtensionSdkDir = "C:\Program Files (x86)\Microsoft SDKs\Windows Kits\10\ExtensionSDKs",
        WindowsLibPath = "References\CommonConfiguration\Neutral",
        WindowsSDKVersion = "\",
        VS120COMNTOOLS = "C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\Tools\",
        VS160COMNTOOLS = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Tools\",
        VSCMD_VER = "16.11.16",
        LIBPATH = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\ATLMFC\lib\x86;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\lib\x86;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\lib\x86\store\references;C:\Windows\Microsoft.NET\Framework\v4.0.30319",
        VCToolsInstallDir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\",
        VSCMD_ARG_app_plat = "Desktop",
        VCIDEInstallDir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\VC\",
        VSCMD_ARG_HOST_ARCH = "x86",
        VCInstallDir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\",
        VSCMD_ARG_TGT_ARCH = "x86",
        VCToolsVersion = "14.29.30133",
        PATH = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\\Extensions\Microsoft\IntelliCode\CLI;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX86\x86;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\VC\VCPackages;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\bin\Roslyn;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Team Tools\Performance Tools;C:\Program Files (x86)\HTML Help Workshop;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\FSharp\Tools;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Tools\devinit;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\\MSBuild\Current\Bin;C:\Windows\Microsoft.NET\Framework\v4.0.30319;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Tools\;C:\Users\11243\xmake;D:\study_software\ctex\install\MiKTeX\miktex\bin\;D:\study_software\ctex\install\UserData\miktex\bin;D:\study_software\ctex\install\MiKTeX\miktex\bin;D:\study_software\ctex\install\CTeX\ctex\bin;D:\study_software\ctex\install\CTeX\cct\bin;D:\study_software\ctex\install\CTeX\ty\bin;D:\study_software\ctex\install\Ghostscript\gs9.05\bin;D:\study_software\ctex\install\GSview\gsview;D:\study_software\ctex\install\WinEdt;C:\Program Files\Git\bin;C:\Program Files\Git\mingw64\libexec\git-core;C:\Program Files\dotnet\;C:\Users\11243\AppData\Local\Programs\Python\Python39\Scripts\;C:\Users\11243\AppData\Local\Programs\Python\Python39\;C:\Users\11243\AppData\Local\Microsoft\WindowsApps;C:\Users\11243\AppData\Local\Programs\Microsoft VS Code\bin;D:\study_software\cloc;C:\Users\11243\AppData\Roaming\npm;C:\Users\11243\.dotnet\tools;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\VC\Linux\bin\ConnectionManagerExe" 
      },
      x64 = {
        VSInstallDir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\",
        LIB = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\ATLMFC\lib\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\lib\x64",
        VCToolsRedistDir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Redist\MSVC\14.29.30133\",
        INCLUDE = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\ATLMFC\include;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include",
        VS110COMNTOOLS = "C:\Program Files (x86)\Microsoft Visual Studio 11.0\Common7\Tools\",
        DevEnvdir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\",
        VisualStudioVersion = "16.0",
        ExtensionSdkDir = "C:\Program Files (x86)\Microsoft SDKs\Windows Kits\10\ExtensionSDKs",
        WindowsLibPath = "References\CommonConfiguration\Neutral",
        WindowsSDKVersion = "\",
        VS120COMNTOOLS = "C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\Tools\",
        VS160COMNTOOLS = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Tools\",
        VSCMD_VER = "16.11.16",
        LIBPATH = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\ATLMFC\lib\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\lib\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\lib\x86\store\references;C:\Windows\Microsoft.NET\Framework64\v4.0.30319",
        VCToolsInstallDir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\",
        VSCMD_ARG_app_plat = "Desktop",
        VCIDEInstallDir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\VC\",
        VSCMD_ARG_HOST_ARCH = "x64",
        VCInstallDir = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\",
        VSCMD_ARG_TGT_ARCH = "x64",
        VCToolsVersion = "14.29.30133",
        PATH = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\\Extensions\Microsoft\IntelliCode\CLI;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\bin\HostX64\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\VC\VCPackages;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\TestWindow;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\TeamFoundation\Team Explorer;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Current\bin\Roslyn;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Team Tools\Performance Tools\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Team Tools\Performance Tools;C:\Program Files (x86)\HTML Help Workshop;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\FSharp\Tools;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Tools\devinit;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\\MSBuild\Current\Bin;C:\Windows\Microsoft.NET\Framework64\v4.0.30319;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\Tools\;C:\Users\11243\xmake;D:\study_software\ctex\install\MiKTeX\miktex\bin\;D:\study_software\ctex\install\UserData\miktex\bin;D:\study_software\ctex\install\MiKTeX\miktex\bin;D:\study_software\ctex\install\CTeX\ctex\bin;D:\study_software\ctex\install\CTeX\cct\bin;D:\study_software\ctex\install\CTeX\ty\bin;D:\study_software\ctex\install\Ghostscript\gs9.05\bin;D:\study_software\ctex\install\GSview\gsview;D:\study_software\ctex\install\WinEdt;C:\Program Files\Git\bin;C:\Program Files\Git\mingw64\libexec\git-core;C:\Program Files\dotnet\;C:\Users\11243\AppData\Local\Programs\Python\Python39\Scripts\;C:\Users\11243\AppData\Local\Programs\Python\Python39\;C:\Users\11243\AppData\Local\Microsoft\WindowsApps;C:\Users\11243\AppData\Local\Programs\Microsoft VS Code\bin;D:\study_software\cloc;C:\Users\11243\AppData\Roaming\npm;C:\Users\11243\.dotnet\tools;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\Common7\IDE\VC\Linux\bin\ConnectionManagerExe" 
      }
    },
    version = "16.0",
    vcvarsall_bat = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvarsall.bat"
  },
  "2012" = {
    vcvarsall = {
      x86 = {
        PATH = "C:\Users\11243\xmake;D:\study_software\ctex\install\MiKTeX\miktex\bin\;D:\study_software\ctex\install\UserData\miktex\bin;D:\study_software\ctex\install\MiKTeX\miktex\bin;D:\study_software\ctex\install\CTeX\ctex\bin;D:\study_software\ctex\install\CTeX\cct\bin;D:\study_software\ctex\install\CTeX\ty\bin;D:\study_software\ctex\install\Ghostscript\gs9.05\bin;D:\study_software\ctex\install\GSview\gsview;D:\study_software\ctex\install\WinEdt;C:\Program Files\Git\bin;C:\Program Files\Git\mingw64\libexec\git-core;C:\Program Files\dotnet\;C:\Users\11243\AppData\Local\Programs\Python\Python39\Scripts\;C:\Users\11243\AppData\Local\Programs\Python\Python39\;C:\Users\11243\AppData\Local\Microsoft\WindowsApps;C:\Users\11243\AppData\Local\Programs\Microsoft VS Code\bin;D:\study_software\cloc;C:\Users\11243\AppData\Roaming\npm;C:\Users\11243\.dotnet\tools",
        VisualStudioVersion = "11.0",
        VS120COMNTOOLS = "C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\Tools\"
      },
      x64 = {
        VS110COMNTOOLS = "C:\Program Files (x86)\Microsoft Visual Studio 11.0\Common7\Tools\",
        PATH = "C:\Users\11243\xmake;D:\study_software\ctex\install\MiKTeX\miktex\bin\;D:\study_software\ctex\install\UserData\miktex\bin;D:\study_software\ctex\install\MiKTeX\miktex\bin;D:\study_software\ctex\install\CTeX\ctex\bin;D:\study_software\ctex\install\CTeX\cct\bin;D:\study_software\ctex\install\CTeX\ty\bin;D:\study_software\ctex\install\Ghostscript\gs9.05\bin;D:\study_software\ctex\install\GSview\gsview;D:\study_software\ctex\install\WinEdt;C:\Program Files\Git\bin;C:\Program Files\Git\mingw64\libexec\git-core;C:\Program Files\dotnet\;C:\Users\11243\AppData\Local\Programs\Python\Python39\Scripts\;C:\Users\11243\AppData\Local\Programs\Python\Python39\;C:\Users\11243\AppData\Local\Microsoft\WindowsApps;C:\Users\11243\AppData\Local\Programs\Microsoft VS Code\bin;D:\study_software\cloc;C:\Users\11243\AppData\Roaming\npm;C:\Users\11243\.dotnet\tools",
        VS120COMNTOOLS = "C:\Program Files (x86)\Microsoft Visual Studio 12.0\Common7\Tools\"
      }
    },
    version = "11.0",
    vcvarsall_bat = "C:\Program Files (x86)\Microsoft Visual Studio 11.0\Common7\Tools\..\..\VC\vcvarsall.bat"
  },
  "2013" = {
    vcvarsall = {
      x86 = {
        VS110COMNTOOLS = "C:\Program Files (x86)\Microsoft Visual Studio 11.0\Common7\Tools\",
        VisualStudioVersion = "12.0",
        PATH = "C:\Users\11243\xmake;D:\study_software\ctex\install\MiKTeX\miktex\bin\;D:\study_software\ctex\install\UserData\miktex\bin;D:\study_software\ctex\install\MiKTeX\miktex\bin;D:\study_software\ctex\install\CTeX\ctex\bin;D:\study_software\ctex\install\CTeX\cct\bin;D:\study_software\ctex\install\CTeX\ty\bin;D:\study_software\ctex\install\Ghostscript\gs9.05\bin;D:\study_software\ctex\install\GSview\gsview;D:\study_software\ctex\install\WinEdt;C:\Program Files\Git\bin;C:\Program Files\Git\mingw64\libexec\git-core;C:\Program Files\dotnet\;C:\Users\11243\AppData\Local\Programs\Python\Python39\Scripts\;C:\Users\11243\AppData\Local\Programs\Python\Python39\;C:\Users\11243\AppData\Local\Microsoft\WindowsApps;C:\Users\11243\AppData\Local\Programs\Microsoft VS Code\bin;D:\study_software\cloc;C:\Users\11243\AppData\Roaming\npm;C:\Users\11243\.dotnet\tools" 
      },
      x64 = {
        VS110COMNTOOLS = "C:\Program Files (x86)\Microsoft Visual Studio 11.0\Common7\Tools\",
        VisualStudioVersion = "12.0",
        PATH = "C:\Users\11243\xmake;D:\study_software\ctex\install\MiKTeX\miktex\bin\;D:\study_software\ctex\install\UserData\miktex\bin;D:\study_software\ctex\install\MiKTeX\miktex\bin;D:\study_software\ctex\install\CTeX\ctex\bin;D:\study_software\ctex\install\CTeX\cct\bin;D:\study_software\ctex\install\CTeX\ty\bin;D:\study_software\ctex\install\Ghostscript\gs9.05\bin;D:\study_software\ctex\install\GSview\gsview;D:\study_software\ctex\install\WinEdt;C:\Program Files\Git\bin;C:\Program Files\Git\mingw64\libexec\git-core;C:\Program Files\dotnet\;C:\Users\11243\AppData\Local\Programs\Python\Python39\Scripts\;C:\Users\11243\AppData\Local\Programs\Python\Python39\;C:\Users\11243\AppData\Local\Microsoft\WindowsApps;C:\Users\11243\AppData\Local\Programs\Microsoft VS Code\bin;D:\study_software\cloc;C:\Users\11243\AppData\Roaming\npm;C:\Users\11243\.dotnet\tools" 
      }
    },
    version = "12.0",
    vcvarsall_bat = "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat"
  }
}

**Comment by waruqi** [comment]
你最近有没有升级过 vs，执行下 `xmake g -c` 后再试试

**Reply by congcong-hehe** [reply]
出问题之前没有更新过vs，出了问题后更新了vs还是没用

**Reply by congcong-hehe** [reply]
还是不行哎..

**Reply by waruqi** [reply]
xmake g -c 试了么

```console
xmake g -c
xmake f -c
xmake
```

**Reply by congcong-hehe** [reply]
PS D:\study_data\pro\mypro\test\xmake\hello> xmake g -c
configure
{
    theme = default
    clean = true
    proxy_pac = pac.lua
    network = public
}
PS D:\study_data\pro\mypro\test\xmake\hello> xmake f -c
checking for platform ... windows
checking for architecture ... x64
checking for Microsoft Visual Studio (x64) version ... 2019
PS D:\study_data\pro\mypro\test\xmake\hello> xmake
[ 25%]: ccache compiling.release src\main.c
error: main.c
src\main.c(1): fatal error C1083: 无法打开包括文件: “stdio.h”: No such file or directory

  > in src\main.c

**Comment by waruqi** [comment]
那就不知道了，你可以看下 `INCLUDE = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\ATLMFC\include;C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include",` 里面有没有 stdio.h ，或者自己找下它在哪。。

感觉还是你这的环境问题，只能你自己排查下了。。

**Reply by congcong-hehe** [reply]
没有哎，只有cstdio，但是找不到stdio.h。stdio.h在windows kits中，找不到windows 10 sdk

**Reply by waruqi** [reply]
那还是你这没装完整。。xmake 是根据 vs 给的 INCLUDE 去找路径，如果这下面没有，那就找不到了。。

**Reply by congcong-hehe** [reply]
我的vs2019中宏WindowsSDK_IncludePath的值是对的呀，hh，我再找找问题

**Reply by waruqi** [reply]
你可以打开  vcprompt cmd ，看下 echo %INCLUDE% 里面的路径有没有带上 stdio.h 的

**Reply by congcong-hehe** [reply]
还是只有这两个目录
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\ATLMFC\include;
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include

**Reply by waruqi** [reply]
那就是你自己的环境问题了，xmake 就是加载的 vsprompt 的 INCLUDE 环境。。你自己捣鼓吧

**Comment by congcong-hehe** [comment, marked as answer]
终于解决了！我在系统里增加了INCLUDE和LIB变量，指向windows kit的路径，然后就好了，谢谢作者大大
